etcdutl/snapshot/v3_snapshot.go

```diff
@@ -311,6 +335,88 @@ func (s *v3Manager) saveDB() error {
 	return nil
 }
 
+// unsafeGetLatestRevision returns the latest revision without locking the given transaction.
+func (s *v3Manager) unsafeGetLatestRevision(tx backend.BatchTx) (latest revision, err error) {
+	err = tx.UnsafeForEach(schema.Key, func(k, _ []byte) (err error) {
+		rev := bytesToRev(k)
+
+		if latest.main < rev.main {
+			latest = rev
+		}
+
+		return nil
+	})
+
+	return
+}
+
+// bumpRevision increases the revision so that, from the client's perspective, the revision never decreases.
+//
+//	If we don't bump the revision, a client looking for a key at one revision finds that the key is actually at a
+//	smaller revision which breaks the promise that revisions never decrease. The client has no way of knowing that
+//	etcd was restored, it only sees the decrease in revision.
+//
+// To ensure that clients don't have a revision that's higher than the restored revision,
+// 'amount' should be a very large number, >= 1_000_000_000.
+//
+// Compaction should be run after this to further help invalidate watcher caches.
```

Comment:
This is incorrect behavior as those flags are independent. Please implement bumpRevision in a way that doesn't require `markLatestRevCompacted` or add a validation that forbids using `bumpRevision` without using `markLatestRevCompacted`.

Reply:
+1. Sounds like `bumpRevision` + `markLatestRevCompacted` should be used together to mark user last observed revision is compacted.

Reply:
Agreed that if we bumpRevision, we want to compact. Can either of you foresee wanting to compact without bumping: should that be an option?